src/app/characters/characters.service.ts

```diff
@@ -21,15 +22,17 @@ export class CharacterService {
 
   constructor(private http: HttpClient, private contextService: ContextService) {}
 
-  getCharacters(term: string): Observable<Character[]> {
-    const options = new HttpParams().set('nameStartsWith', term);
+  getCharacters(term: string): Observable<MarvelResponse['data']> {
```

Comment:
The `MarvelResponse['data']` type looks weird. I think that the extra interface for the `data` property would make a good fit in this case.

Reply:
What do you think about creating a more generic interface in order to reuse it at `ComicService`?
```ts
interface MarvelResponseData<T> {
  total: number;
  results: T[];
}

interface MarvelResponse<T> {
  attributionText: string;
  data: MarvelResponseData<T>;
} 
```
However, I'm not sure in which directory to create the new file of this generic model. What do you think?

Reply:
That is a great idea @nikosanif 😀 I really like the approach using generics 👌 Typically, I use the `core` folder for this type of file, similar to the `character.model.ts` file.

Reply:
Done.